scio-elasticsearch/es6/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -415,32 +456,47 @@ public void setup() throws Exception {
       public void processElement(ProcessContext c) throws Exception {
         final Iterable<T> values = c.element().getValue();
 
-        // Elasticsearch throws ActionRequestValidationException if bulk request is empty,
-        // so do nothing if number of actions is zero.
-        if (!values.iterator().hasNext()) {
```

Comment:
we can still do an early get out here and avoid the allocations bellow, right?

Reply:
Yeah, that's true. Should I leave this block in then?

Reply:
yeah let's keep it.